### PR TITLE
Adding new simple verifier for decoupling

### PR DESCRIPTION
**Pull request by coax1d**
This is the start of the migration to a new simple verifier as discussed in #27 this includes the following tasks:
- [x] migrate old verifiers to new `SimpleVerifier` trait.
- [x] update `Executive` logic
- [x] Migrate old tests and old code depending old verifier logic

### CODE REVIEW

**Comment by JoshOrndorff** [comment]
Overall, I totally agree with the direction of this PR. I opened #42 which makes the macro unnecessary.

**Comment by coax1d** [comment]
@JoshOrndorff  let me know what you think if this is good for now. Thanks!!

File: tuxedo-template-runtime/src/kitties.rs [hunk 1]
```diff
@@ -559,4 +543,144 @@ mod test {
     fn check_child_num_breedings_non_zero_fails() {
         // TODO
     }
+
+    #[test]
+    fn breed_wrong_input_type_fails() {
```
Comment (coax1d) [inline tuxedo-template-runtime/src/kitties.rs:548]: Not sure why these tests are popping up when they already exist on main

File: tuxedo-template-runtime/src/kitties.rs [hunk 2]
```diff
@@ -1,34 +1,18 @@
...
-//! In order to submit a valid transaction you must strutucture it as follows:
-//! 1.) Input must contain 1 mom and 1 dad
-//! 2.) Output must contain Mom, Dad, and newly created Child
-//! 3.) A child's DNA is calculated by:
-//!         BlakeTwo256::hash_of(MomDna, DadDna, MomCurrNumBreedings, DadCurrNumberBreedings)
-//!
-//! There are a only a finite amount of free breedings available before it starts to cost money
-//! to breed kitties.
-//!
-
```
Comment (JoshOrndorff) [inline tuxedo-template-runtime/src/kitties.rs:18]: Is this deletion intentional?

File: tuxedo-core/src/verifier.rs [hunk 3]
```diff
@@ -29,10 +33,54 @@ pub trait Verifier: Debug + Encode + Decode + Clone {
...
+            .collect();
+
+        // Call the simple verifier
+        SimpleVerifier::verify(self, &input_data, &output_data)
+    }
+}
+
+// A trivial verifier that verifies everything. Not practical. More for testing
+// and for the sake of making things compile before I get around to writing the
+// amoeba nd PoE verifiers
```
Comment (JoshOrndorff) [inline tuxedo-core/src/verifier.rs:81]:
```suggestion
/// Utilities for writing verifier-related unit tests
```

File: tuxedo-template-runtime/src/lib.rs [hunk 4]
```diff
@@ -295,26 +294,26 @@ impl From<runtime_upgrade::VerifierError> for OuterVerifierError {
...
         &self,
-        input_data: &[DynamicallyTypedData],
-        output_data: &[DynamicallyTypedData],
+        inputs: &[Output<R>],
+        outputs: &[Output<R>],
     ) -> Result<TransactionPriority, OuterVerifierError> {
         Ok(match self {
-            Self::Money(money) => money.verify(input_data, output_data)?,
-            Self::FreeKittyVerifier(free_breed) => free_breed.verify(input_data, output_data)?,
+            Self::Money(money) => Verifier::verify(money, inputs, outputs)?,
```
Comment (JoshOrndorff) [inline tuxedo-template-runtime/src/lib.rs:303]: I'm surprised this change is necessary. Of course we are using the new, more general, `Verifier` trait, but I didn't expect that we should have to explicitly mention it as long as the `SimpleVerifier` is not in scope.

**Review by JoshOrndorff — APPROVED** [review]
Looks basically good. I made a few suggestions, and we need the CI green. Then I'm fine with merging.

File: tuxedo-template-runtime/src/lib.rs (hunk 4 above)
Comment (coax1d) [reply tuxedo-template-runtime/src/lib.rs:303]: You are correct I had initially the `SimpleVerifier` in scope. Am changing

File: tuxedo-template-runtime/src/kitties.rs (hunk 2 above)
Comment (coax1d) [reply tuxedo-template-runtime/src/kitties.rs:18]: Nope thx for catching I made this PR in a rush.

**Pull request merged**